### PR TITLE
🌱 Migrate test/e2e/shared/openstack.go to structured logging

### PR DESCRIPTION
**Pull request by sophiajwitt**
**What this PR does / why we need it**:

Updates logging formatting as outlined in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md and adds verbosity levels.

/hold

### CODE REVIEW

File: test/e2e/shared/openstack.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ type ServerExtWithIP struct {
 
 // ensureSSHKeyPair ensures A SSH key is present under the name.
 func ensureSSHKeyPair(e2eCtx *E2EContext) {
-	Logf("Ensuring presence of SSH key %q in OpenStack", DefaultSSHKeyPairName)
+	logger.V(2).Info("Ensuring presence of SSH key in OpenStack", "Key pair name", DefaultSSHKeyPairName)
```
Comment (mdbooth) [inline test/e2e/shared/openstack.go:67]: This is in test so not very important, but if you respin this I'd probably say 'presence of keypair', and "name".

File: test/e2e/shared/openstack.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func ensureSSHKeyPair(e2eCtx *E2EContext) {
 	}
 
 	sshDir := filepath.Join(e2eCtx.Settings.ArtifactFolder, "ssh")
-	Logf("Storing keypair in %q", sshDir)
+	logger.V(2).Info("Storing keypair in ssh directory", "ssh directory", sshDir)
```
Comment (mdbooth) [inline test/e2e/shared/openstack.go:87]: Again not super important, but I'd probably change "ssh directory" to "directory". I don't think we typically have spaces in parameter names.

**Review by mdbooth — COMMENTED** [review]
/approve

Some nits if you respin.

**Review by dulek — CHANGES_REQUESTED** [review]
Looks like you've added `controllers/openstackcluster_controller.go` by mistake here.

**Pull request closed without merging**